chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -254,8 +254,12 @@ def self.all(node)
           node[:block_device].keys.map{|d|Disk.new(node,d)}
         end
 
-        def self.unclaimed(node)
+        def self.unclaimed(node, include_mounted=false)
           all(node).select do |d|
+            unless include_mounted
+              %x{lsblk #{d.name} --noheadings --output MOUNTPOINT | grep -q -v ^$}
```

Comment:
you could use ohai's data to exclude mounted filesystems. That way, the code would be later shareable with https://github.com/crowbar/barclamp-crowbar/blob/master/crowbar_framework/app/models/node_object.rb#L606 (see also https://trello.com/c/eBwdpumX/425-fix-code-duplication-in-crowbar-and-cookbooks). But I'm also fine with the current code...